### PR TITLE
feat: [CDS-45142]: 1. Add test for Freeze Studio Wrapper, 2. Test cases For NewFreezeWindow Button

### PR DESCRIPTION
**Pull request by ishantmehtaharness**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by vkbansal-harness** [comment]
retrigger cypress-cd

**Review by vkbansal-harness — APPROVED** [review]

File: src/modules/72-freeze-windows/components/__tests__/FreezeStudioWrapper.test.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  useUpdateFreeze: jest.fn().mockImplementation(() => ({})),
+  useGetProjectList: jest.fn().mockImplementation(() => ({})),
+  useGetServiceList: jest.fn().mockImplementation(() => ({ loading: false, data: serviceData, refetch: jest.fn() }))
+}))
+
+describe('Freeze Window Studio Wrapper', () => {
+  test('it should render FreezeWindow Studio in a wrapper', () => {
+    const { container, getByText } = render(
+      <TestWrapper
+        path="/account/:accountId/:module/orgs/:orgIdentifier/projects/:projectIdentifier/setup/freeze-window-studio/window/:windowIdentifier/"
```
Comment (udayvunnam) [inline src/modules/72-freeze-windows/components/__tests__/FreezeStudioWrapper.test.tsx:34]: Let's get this route from actual route declarations `routes.toFreezeWindowStudio({ ...projectPathProps, ...modulePathProps })`

File: src/modules/72-freeze-windows/components/__tests__/FreezeStudioWrapper.test.tsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    )
+
+    // header assertion
+    expect(document.getElementsByClassName('Breadcrumbs--breadcrumb').length).toBe(2)
+    expect(document.getElementsByClassName('freezeToggler')[0]).toBeDefined()
+    expect(document.getElementsByClassName('visualYamlToggle')[0]).toBeDefined()
+
+    // Should have tab list, and have 3 tabs
+    expect(document.getElementsByClassName('bp3-tab-list')[0]).toBeDefined()
+    expect(document.getElementsByClassName('bp3-tab').length).toBeDefined()
```
Comment (udayvunnam) [inline src/modules/72-freeze-windows/components/__tests__/FreezeStudioWrapper.test.tsx:48]: lookout for accessible names or actual text to query by.

File: src/modules/72-freeze-windows/components/__tests__/FreezeStudioWrapper.test.tsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+    // Overview section should be rendered by default
+    expect(getByText('overview')).toBeDefined()
+    expect(getByText('freezeWindows.freezeStudio.freezeConfiguration')).toBeDefined()
+    expect(getByText('common.schedule')).toBeDefined()
+    expect(getByText('freezeWindows.freezeStudio.freezeOverview')).toBeDefined()
+
+    // Notifications
+    expect(getByText('notifications.pipelineName')).toBeDefined()
+    expect(container).toMatchSnapshot('Freeze Studio Wrapper Snapshot')
```
Comment (udayvunnam) [inline src/modules/72-freeze-windows/components/__tests__/FreezeStudioWrapper.test.tsx:58]: We are taking snapshot in the page's default state. Is this required

File: src/modules/72-freeze-windows/components/__tests__/NewFreezeWindowButton.test.tsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        </FreezeWindowContext.Provider>
+      </TestWrapper>
+    )
+
+    expect(container).toMatchSnapshot('New Freeze Window Button')
+
+    // click on button
+    const button = container.getElementsByTagName('button')
+    expect(button[0]).toBeDefined()
+    fireEvent.click(button[0])
```
Comment (udayvunnam) [inline src/modules/72-freeze-windows/components/__tests__/NewFreezeWindowButton.test.tsx:37]: action without further assertion. Is this intentional? Also, prefer userEvent

**Review by pankajsingh-harness — APPROVED** [review]

File: src/modules/72-freeze-windows/components/__tests__/FreezeStudioWrapper.test.tsx (hunk 1 above)
Comment (ishantmehtaharness) [reply src/modules/72-freeze-windows/components/__tests__/FreezeStudioWrapper.test.tsx:34]: Yes, this sounds good
But in test case, I want to assert for route path as well.

File: src/modules/72-freeze-windows/components/__tests__/FreezeStudioWrapper.test.tsx (hunk 3 above)
Comment (ishantmehtaharness) [reply src/modules/72-freeze-windows/components/__tests__/FreezeStudioWrapper.test.tsx:58]: Yes, I am doing one complete snapshot test of the DOM tree

File: src/modules/72-freeze-windows/components/__tests__/NewFreezeWindowButton.test.tsx (hunk 4 above)
Comment (ishantmehtaharness) [reply src/modules/72-freeze-windows/components/__tests__/NewFreezeWindowButton.test.tsx:37]: Yes, this is checking if the button is actually clicked
I will add one more assertion here, if the route path is also changed with this. will do in next PR

**Review by udayvunnam — APPROVED** [review]

**Pull request merged**